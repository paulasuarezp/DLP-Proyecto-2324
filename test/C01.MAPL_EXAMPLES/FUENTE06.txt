// Imprimir 20 los primeros n√∫meros de la serie de fibonacci

int penultimo, ultimo, actual;
int imprimidos;

print 0;
print 1;
imprimidos = 2;

penultimo = 0;
ultimo = 1;

while (imprimidos < 20) {
    actual = ultimo + penultimo;

    println actual;
    imprimidos = imprimidos + 1;

    penultimo = ultimo;
    ultimo = actual;
}
