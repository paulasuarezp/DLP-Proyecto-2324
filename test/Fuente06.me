//*
Ejemplo Fuente06.
Imprimir 20 los primeros números de la serie de fibonacci
*//

class Fuente06;
	global
		vars
			penultimo, ultimo, actual : INTEGER;
			imprimidos : INTEGER;
			
	create
		main; 
			
	feature	main is
		/* local

		do
			println 0;
			println 1;
			imprimidos := 2;

			from
				penultimo := 0;
				ultimo := 1;

			until imprimidos >= 20 loop /* while (imprimidos < 20)
				actual := ultimo + penultimo;

				println actual;
				imprimidos := imprimidos + 1;

				penultimo := ultimo;
				ultimo := actual;
			end
	end /*main
	
end /* Fuente06

run main();