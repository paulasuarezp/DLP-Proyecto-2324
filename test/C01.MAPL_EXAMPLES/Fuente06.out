' Compilador mini Eiffel.
' Aquilino A. Juan.
' Versi√≥n 2024.2 (c)
#source "Fuente06.me"
#global penultimo: int
#global ultimo: int
#global actual: int
#global imprimidos: int
#func main
#ret void
CALL main
HALT

#line 15
main:

#line 19
PUSHI 0
OUTI
PUSHB 10
OUTB

#line 20
PUSHI 1
OUTI
PUSHB 10
OUTB

#line 21
PUSHA 6
PUSHI 2
STOREI

#line 24
' inicio from_0000

#line 24
PUSHA 0
PUSHI 0
STOREI

#line 25
PUSHA 2
PUSHI 1
STOREI

#line 27
UntilCond_0000:
PUSHA 6
LOADI
PUSHI 20
GEI
JNZ UntilEnd_0000

#line 28
PUSHA 4
PUSHA 2
LOADI
PUSHA 0
LOADI
ADDI
STOREI

#line 30
PUSHA 4
LOADI
OUTI
PUSHB 10
OUTB

#line 31
PUSHA 6
PUSHA 6
LOADI
PUSHI 1
ADDI
STOREI

#line 33
PUSHA 0
PUSHA 2
LOADI
STOREI

#line 34
PUSHA 2
PUSHA 4
LOADI
STOREI
JMP UntilCond_0000
UntilEnd_0000:

#line 34
RET 0, 0, 0
